lib/taskana-core/src/main/java/pro/taskana/spi/routing/api/TaskRoutingProvider.java

```diff
@@ -14,10 +25,12 @@ public interface TaskRoutingProvider {
   void initialize(TaskanaEngine taskanaEngine);
 
   /**
-   * Determines a WorkbasketId for a given task.
+   * Determines the {@linkplain Workbasket#getId() id} of the {@linkplain Workbasket} for a given
+   * {@linkplain Task}.This method will be invoked by TASKANA when it is asked to create a
+   * {@linkplain Task} that has no {@linkplain Workbasket} assigned.
    *
-   * @param task {@linkplain Task} The task for which a workbasket must be determined.
-   * @return the id of the workbasket in which the task is to be created.
+   * @param task the {@linkplain Task} for which a {@linkplain Workbasket} must be determined
+   * @return the {@linkplain Workbasket#getId() id} of the {@linkplain Workbasket}
    */
   String determineWorkbasketId(Task task);
```

Comment:
I am missing:
* link to TaskService#create
* null return if no routing should take place
* if multiple service provider are defined they either have to return null or the return value has to match. If more than 1 WorkbasketId gets returned an error log will be created.